### ISSUE TITLE
False positive with inputs inside labels

### ISSUE THREAD

**Issue by jeremyZX** [inclusive-design/AChecker#106]
In the algorithm for [Rule 204 - All input elements, type of "radio" have a label containing text](https://achecker.ca/checker/suggestion.php?id=204), steps 7 and 8 state:

7. The ``input`` element is contained by a ``label`` element
8. Check if the label contains text

However, your detection for text inside the label is limited to text preceding the input. So the following element will validate as valid:

```html
<label>
    My billing and shipping addresses are the same <input name="billing" type="radio" />
</label>
```

But this element will not:

```html
<label>
    <input name="billing" type="radio" /> My billing and shipping addresses are the same
</label>
```

Since [the WAI recommends that all left-to-right languages place text to the right of radio buttons and checkboxes](https://www.w3.org/WAI/tutorials/forms/labels/#visual-position-of-label-text), in LTR languages, this check results in a false positive.

The problem appears to be located in the ``associated_label_has_text`` function, which constructs a regular expression for the innerHTML of the label element, but omits a second capture group for text / HTML nodes to the right of the input element.  So a rudimentary fix would look something like the following:

```php
  public static function associated_label_has_text($e, $content_dom)
  {
      ...

      // 2. The element $e is contained by a "label" element
      if ($e->parent()->tag == "label")
      {
        $pattern = "/(.*)". preg_quote($e->outertext, '/') ."(.*)/";
        preg_match($pattern, $e->parent->innertext, $matches);
        if (strlen(trim($matches[1])) > 0) return true;
        if (strlen(trim($matches[2])) > 0) return true;
      }

      ...

      return false;
}
```

**Comment by cindyli** [comment]
Thanks for reporting the issue in such a detail, @jeremyZX. We'll have a look at the next development cycle.
